hcldec/spec.go

```diff
@@ -1565,6 +1565,58 @@ func (s *TransformFuncSpec) sourceRange(content *hcl.BodyContent, blockLabels []
 	return s.Wrapped.sourceRange(content, blockLabels)
 }
 
+// ValidateFuncSpec is a spec that allows for extended
+// developer-defined validation. The validation function receives the
+// result of the wrapped spec.
+//
+// The Subject field of the returned Diagnostic is optional. If not
+// specified, it is automatically populated with the range covered by
+// the wrapped spec.
+//
+type ValidateFuncSpec struct {
```

Comment:
The `Func` in `TransformFunc` is meaning the `function.Function` (`cty`) sense of that word, rather than the Go function sense of the word.

I suggest we name this just `ValidateSpec`, which is a very subtle distinction but subjectively it feels like it'd be clearer to someone scanning the docs what this spec is for, and it also leaves the door open for us to potentially have `ValidateExprSpec` and `ValidateFuncSpec` corresponding with the `Transform...` ones of the same name, if reasons for those arise later. 

(I could see `ValidateExprSpec` showing up if we implement a corresponding feature inside the `hcldec` command line tool, for example, because that includes an HCL-based language for writing `hcldec` specs. That can wait until we have a use-case for it, though!)

Reply:
👍  done!